src/openrct2-ui/windows/Options.cpp

```diff
@@ -1789,22 +1802,28 @@ static void window_options_invalidate(rct_window* w)
             window_options_audio_widgets[WIDX_TITLE_MUSIC].text = window_options_title_music_names[gConfigSound.title_music];
 
             widget_set_checkbox_value(w, WIDX_SOUND_CHECKBOX, gConfigSound.sound_enabled);
+            widget_set_checkbox_value(w, WIDX_MASTER_SOUND_CHECKBOX, gConfigSound.master_sound_enabled);
             widget_set_checkbox_value(w, WIDX_MUSIC_CHECKBOX, gConfigSound.ride_music_enabled);
             widget_set_checkbox_value(w, WIDX_AUDIO_FOCUS_CHECKBOX, gConfigSound.audio_focus);
 
             // Initialize only on first frame, otherwise the scrollbars wont be able to be modified
             if (w->frame_no == 0)
             {
+                widget = &window_options_audio_widgets[WIDX_MASTER_VOLUME];
+                w->scrolls[0].h_left = ceil(
+                    (gConfigSound.master_volume / 100.0f) * (w->scrolls[0].h_right - ((widget->right - widget->left) - 1)));
```

Comment:
Please split this calculation in two parts and store them, so make it more readable and make debugging easier.
`(gConfigSound.master_volume / 100.0f)` and `(w->scrolls[0].h_right - ((widget->right - widget->left) - 1))`.

Reply:
I left this in the same style as the other two scrolls. I will refactor this, but there are many other similar cases in the existing code that are near unreadable - a full refactor would be more appropriate sometime.

Reply:
That's definitely true, and small refactors like this makes it much easier in the long run.